src/vm_cpu.rs

```diff
@@ -149,6 +160,120 @@ impl VmPmuState {
             None
         }
     }
+
+    fn gen_counter_mask(high: u64, low: u64) -> u64 {
```

Comment:
Possibly, but doesn't that require an extra check for high/low underflow?

Reply:
It's a private function, so leave it to the caller? Seems silly to have these extra branches when it can be trivially implemented with a couple bitops.